### ISSUE TITLE
UX refinement: can Shortcut Help intercept and describe keystrokes that access top menus on Windows?

### ISSUE THREAD

**Issue by ScottChesworth** [jcsteh/osara#276]
We're moving toward unblocking access to all top menus in the next revision of the key map. Just now, I noticed that executing these newly freed up keystrokes accesses the menu system even when Shortcut Help is enabled. If possible, I think a smoother UX would be for OSARA to report which menu each keystroke will access, without performing the task. This would be a Windows=only thing seeing as access to the top menus doesn't work the same way on Mac. Here's a list of keystrokes followed by suggested reports:

Alt+F: File menu
Alt+E: Edit menu
Alt+V: View menu
Alt+I: Insert menu
Alt+M: Item menu
Alt+T: Track menu
Alt+O: Options menu
Alt+X: Extensions menu
Alt+H: Help menu

The two holdouts right now are Alt+X and Alt+M. The Key Map Consortium group is working on finding new homes for those.

Thoughts? Is there a graceful way to handle this that isn't a ton of grunt work?

**Comment by jcsteh** [comment]
I think it's possible, but not trivial. However, given that menu bars are a
standard Windows thing, users should already be pretty familiar with how to
explore the menu bar, work out the access keys, etc. So, I think this is
low priority at best.

Implementation note: This would need to be done in keyboardHookProc. For
any letter pressed with alt as the only modifier, the code would need to
search the menus in the menu bar, looking for a menu with that access key.
I think the text for the menus will have "&" before the access key; e.g.
"&File", so we'd search for "&f". One problem is that if we intercept that
key, I think releasing alt will then cause the menu bar to pop up, so the
code would probably need to inject a 0xff vk code to prevent that.

**Comment by ScottChesworth** [comment]
I've tutored a few folks who are starting from ground zero with DAWs and aren't super confident Windows users yet. Unexpectedly ending up in menus seemed to make them less trusting of Shortcut Help being a safe way to explore, so figured it was worth filing. Agreed on it being low priority though.

**Comment by ScottChesworth** [comment]
And that was the wrong button. *sigh* maybe I need a Shortcut Help for GitHub. :)

**Comment by RDMurray** [comment]
It would also be good if users could assign those shortcuts in their own keymaps.  I don't know how osara could tell if the key has been assigned to something else before the keystroke is intercepted.

P.S is it really necessary to free up these keys?  Menus can always be accessed by pressing and releasing alt, then pressing the corresponding letter.

**Comment by jcsteh** [comment]
> It would also be good if users could assign those shortcuts in their own keymaps.

I really feel this is out of scope for OSARA. That is functionality which isn't really related to screen reader accessibility. Also, that would be overriding core functionality of the standard Windows menu bar. I guess I can see why users might want to do this, but regardless, it doesn't belong in OSARA.

**Comment by ScottChesworth** [comment]
Hi @RDMurray , I think freeing 'em up is a good shout because 1) it's been one of the more prevalent requests since the last big key map revision, 2) I feel like I'm getting stuff done faster since trying it, and 3) I've also been noticing some weird behaviour with the Alt key sneaking in in recent Reaper versions. When I switch back into Reaper from another app, the first press of Alt doesn't seem to have any effect. Alt in combination with a letter works correctly on the first try here.

In the meantime, I'll chase the weird behaviour mentioned above with Cockos.

**Comment by ScottChesworth** [comment]
FYI, the first press of Alt yielding no result bug mentioned in my previous comment has been resolved by Cockos.
